src/types/input.jl

```diff
@@ -2,81 +2,151 @@
 
 abstract type InputType end
 
-struct Bosonic <:InputType
-end
-struct PartDist <:InputType
-end
-abstract type ToyModel <: InputType
-end
-abstract type RandomModel <: InputType
-end
-struct Distinguishable <:InputType
+abstract type Bosonic <: InputType end
+
+abstract type GaussianState <: Bosonic end
```

Comment:
bosonic is meant to say perfectly indistinguishable so this subtyping shouldn't be